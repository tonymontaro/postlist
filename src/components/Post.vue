<template>
  <li class="post">
    Post {{ post.id }}
    <br />
    <button v-if="index > 0" @click="() => moveUp(index)">up</button>
    <button v-if="!isLastPost" @click="() => moveDown(index)">down</button>
  </li>
</template>

<script>
export default {
  name: "post",
  props: ["index", "post", "moveUp", "moveDown", "isLastPost"]
};
</script>
