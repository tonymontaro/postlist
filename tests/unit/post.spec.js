import { mount } from "@vue/test-utils";

import Post from "@/components/Post";

describe("Post Component", () => {
  test("renders post item with up and down links", () => {
    const wrapper = mount(Post, {
      propsData: {
        key: 1,
        post: { id: 1, title: "Title" },
        index: 3,
        moveUp: () => {},
        moveDown: () => {},
        isLastPost: false
      }
    });

    expect(wrapper.findAll("a")).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
  });
});
