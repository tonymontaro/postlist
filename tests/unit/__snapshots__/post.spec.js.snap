// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component renders post item with up and down links 1`] = `
<li class="p-3 py-5 bg-white shadow-lg rounded-sm white-box w-full mb-5 relative pr-10" key="1">
  1. Title
  <a href="#" class="chevron-down absolute"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" icon="CheveronDown" class="w-5 absolute ">
      <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"></path>
    </svg></a> <a href="#" class="chevron-up absolute"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" icon="CheveronUp" class="w-5 absolute ">
      <path d="M10.707 7.05L10 6.343 4.343 12l1.414 1.414L10 9.172l4.243 4.242L15.657 12z"></path>
    </svg></a></li>
`;
